src/partners/thorchain.ts

```diff
@@ -147,19 +148,43 @@ const makeThorchainPlugin = (info: ThorchainInfo): PartnerPlugin => {
         if (txStatus !== 'success') {
           continue
         }
-        if (pools.length !== 2) {
+
+        // Find the source asset
+        let srcAsset: string | undefined
+        let depositAmount: number | undefined
+        let srcAddress: string | undefined
+        for (const txIn of txIns) {
+          for (const coin of txIn.coins) {
+            if (pools.includes(coin.asset)) {
+              srcAsset = coin.asset
+              depositAmount = Number(coin.amount) / THORCHAIN_MULTIPLIER
+              srcAddress = txIn.address
+              break
+            } else {
+              if (pools.length === 1) {
```

Comment:
It doesn't really seem to make sense to have this condition in the `txIn.coins` loop.